### PR TITLE
chore(deps): replace `babel-eslint` with `@babel/eslint-parser`

### PR DESCRIPTION
**Pull request by marcusrbrown**
The `babel-eslint` package was deprecated at v10.1.0 and
moved to `@babel/eslint-parser`.

## Description

Replaced the deprecated `babel-eslint` dependency with `@babel/eslint-parser`. Added the `requireConfigFile` flag to the `parserOptions` key and disabled the requirement.

Please make sure that the PR fulfills these requirements

- [x] I checked that there aren't other open Pull Requests for the same update/change.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] Rule changes includes a comment to describe the reasoning behind the change.
- [ ] PR contains a single rule change.
    
## Motivation and Context

The `babel-eslint` package was deprecated at v10.1.0 and moved to the Babel monorepo as `@babel/eslint-parser`. This package must not depend on deprecated packages and should stay current with the Babel ecosystem.

## How Has This Been Tested?

I ran `npm  ci`, followed by `npm test`.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [x] Dependency update

### CODE REVIEW

**Comment by marcusrbrown** [comment]
Looks like this will close #93.

**Review by 10xLaCroixDrinker — COMMENTED** [review]
Why such large changes to the lockfile?

**Comment by marcusrbrown** [comment]
> Why such large changes to the lockfile?

I couldn't say specifically. There was a lot of refactoring done for Babel monorepo projects and replacing the older package would have brought in many current dependencies. Deps in `eslint-config-amex` are years out of date. I planned to leave `npm audit` on a separate PR. I'm blocked on work items until we can get this package up to date.

**Review by 10xLaCroixDrinker — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
     "node": ">= 12.0.0"
   },
   "dependencies": {
-    "babel-eslint": "^10.1.0",
+    "@babel/eslint-parser": "7.16.5",
```
[the remark this answers is not in the document]
Comment (marcusrbrown) [reply package.json:45]: We use Babel v7 in our stack, unless I am mistaken. This package is part of Babel v7. At the top of the README there is a notice describing the transition: https://github.com/babel/babel-eslint. At the earliest point I could find in the Babel repo history, `@babel/eslint-parser` had moved to v7 of Babel. If they are keeping in line with semver there shouldn't be any breaking changes.

**Review by Matthew-Mallimo — APPROVED** [review]

**Pull request merged**